dss/api/subscriptions_v2.py

```diff
@@ -39,6 +40,11 @@ def find(replica: str):
 
 @security.authorized_group_required(['hca', 'public'])
 def put(json_request_body: dict, replica: str):
+    owner = security.get_token_email(request.token_info)
+    if len([s for s in get_subscriptions_for_owner(Replica[replica], owner) if owner == s['owner']]):
```

Comment:
Instead of retrieving the entire list of subscriptions, please look into `Select COUNT`: https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html#DDB-Query-request-Select

This may require extending the DSS DynamoDB interface, as well as the v2 subscriptions interface.